src/main/java/com/palominolabs/metrics/newrelic/CustomMetricAttributeFilter.java

```diff
@@ -0,0 +1,466 @@
+ *          TIMER_MIN: false
+ *     metricName2:
+ *          COUNTER_COUNT: true
+ * </pre>
+ *
+ * Both constructors receives a {@link MetricAttributeFilter} which will be used as fallback for all metrics not specified in configuration.
+ * In case <i>fallback</i> is null, {@link AllDisabledMetricAttributeFilter} will be used.
+ */
+public class CustomMetricAttributeFilter implements MetricAttributeFilter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CustomMetricAttributeFilter.class);
+
+    private Table<String, NewRelicMetric, Boolean> enabledMetrics;
+    private MetricAttributeFilter fallback;
+
+    /**
+     * @param metricConfig table containing metrics config
+     * @param fallback to be used when there metrics config has no entry for the metric to be reported
+     */
+    public CustomMetricAttributeFilter(Table<String, NewRelicMetric, Boolean> metricConfig, @Nullable MetricAttributeFilter fallback) {
+        Preconditions.checkArgument(metricConfig != null, "metricConfig cannot be null");
+        this.enabledMetrics = metricConfig;
+        this.fallback = getFallbackMetricFilter(fallback);
+    }
+
+    /**
+     * @param metricConfigURI URI to a yaml file containing metrics config
+     * @param fallback to be used when there metrics config has no entry for the metric to be reported
+     */
+    public CustomMetricAttributeFilter(URI metricConfigURI, @Nullable MetricAttributeFilter fallback) {
+        Preconditions.checkArgument(metricConfigURI != null, "metricConfigURI cannot be null");
+        this.enabledMetrics = loadMetrics(metricConfigURI);
+        this.fallback = getFallbackMetricFilter(fallback);
+    }
+
+    private Table<String, NewRelicMetric, Boolean> loadMetrics(URI yamlSource) {
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+
+        try {
+            Map<String, Map<NewRelicMetric, Boolean>> o = mapper
+                    .readValue(yamlSource.toURL(), new TypeReference<Map<String, Map<NewRelicMetric, Boolean>>>() {});
+            return table(o);
+        } catch (IOException e) {
+            LOG.error("Could not parse NewRelic metrics config file. Check expected format", e);
+            throw new IllegalArgumentException("NewRelic metrics config file has wrong format", e);
+        }
+    }
+
+    private MetricAttributeFilter getFallbackMetricFilter(MetricAttributeFilter fallback) {
+        return fallback == null ?
+                new AllDisabledMetricAttributeFilter() :
+                fallback;
+    }
+
+    private Boolean isEnabledWithFallback(final String name, NewRelicMetric newRelicMetric, Supplier<Boolean> fallbackSupplier) {
+        return Optional.fromNullable(enabledMetrics.get(name, newRelicMetric)).or(fallbackSupplier);
+    }
+
+    @Override
+    public boolean recordTimerMin(final String name, final Timer metric) {
+        return isEnabledWithFallback(name, NewRelicMetric.TIMER_MIN, new Supplier<Boolean>() {
```

Comment:
*sigh* Given that I already had an issue a while back asking for Java 1.6 support, I suppose I shouldn't prompt you to use Java 8's BooleanSupplier here :(